src/SonarScanner.MSBuild.PreProcessor/HostInfo.cs

```diff
@@ -98,9 +100,13 @@ public record CloudHostInfo(string ServerUrl, string ApiBaseUrl, string Region)
         }
     }
 
-    public static bool IsKnownUrl(string url)
+    public static bool IsKnownUrl(string url) =>
+        KnownRegions.Any(x => UrlEquals(x.ServerUrl, url));
+
+    private static bool UrlEquals(string url1, string url2)
     {
-        url = url.TrimEnd('/');
-        return KnownRegions.Any(x => x.ServerUrl.Equals(url, StringComparison.InvariantCultureIgnoreCase));
+        url1 = url1?.TrimEnd('/');
```

Comment:
> I don't see how the coverage can be bumped and I also don't want to remove the conditional.

As we have `x.ServerUrl` always as the first argument, the `?` can be safely removed. To make the code more regular, you can make this an instance method with just one `url` parameter. Then then, it will be used as `x.UrlEquals(url)`, because `url1` will be directly `ServerUrl`